src/scripts/main.js

```diff
@@ -1,11 +1,41 @@
 'use strict';
 
 function waitFor(element, eventName) {
-  // write your code here
+  return new Promise((resolve, reject) => {
+    element.addEventListener('click', (e) => {
+      if (e.target === element) {
+        resolve(
+          `It was ${eventName} on the element: ${element.nodeName}, id: ${element.id}.`,
+        );
+      }
+    });
+
+    element.addEventListener('contextmenu', (e) => {
+      if (e.target === element) {
+        resolve(
+          `It was ${eventName} on the element: ${element.nodeName}, id: ${element.id}.`,
+        );
+      }
+    });
+
+    element.addEventListener('blur', (e) => {
+      if (e.target === element) {
+        resolve(
+          `It was ${eventName} on the element: ${element.nodeName}, id: ${element.id}.`,
+        );
+      }
+    });
+  });
```

Comment:
The `waitFor` function is designed to listen for 'click', 'contextmenu', and 'blur' events, but it is being used to listen for 'input' events on lines 49 and 50. This will not work as intended because 'input' events are not being handled in the `waitFor` function. Consider adding an event listener for 'input' events if needed.